File: baekjoon_all/11000+/boj_11723.cpp
// Solve 2022-09-06
// Update 2023-08-16

#include <bits/stdc++.h>
using namespace std;

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;

int main() {
    FASTIO;

    int m;
    cin >> m;

    int st = 0;

    for (int i = 0; i < m; i++) {
        string cmd;
        cin >> cmd;

        if (cmd == "add") {
            int x;
            cin >> x;

            st |= 1 << x;
        }
        else if (cmd == "remove") {
            int x;
            cin >> x;

            st &= ~(1 << x);
        }
        else if (cmd == "check") {
            int x;
            cin >> x;

            if (st & (1 << x)) {
                cout << 1 << '\n';
            }
            else {
                cout << 0 << '\n';
            }
        }
        else if (cmd == "toggle") {
            int x;
            cin >> x;

            st ^= 1 << x;
        }
        else if (cmd == "all") {
            st = 2097151;
        }
        else {
            st = 0;
        }
    }

    return 0;
}

File: baekjoon_all/11000+/boj_11723.py
# Solve 2025-04-27

import sys
input = lambda: sys.stdin.readline().rstrip()

li = [0 for _ in range(21)]

for _ in range(int(input())):
    cmd = input()

    if cmd == "all":
        li = [1 for _ in range(21)]
        continue

    if cmd == "empty":
        li = [0 for _ in range(21)]
        continue

    cmd, x = cmd.split()
    x = int(x)

    if cmd == "add":
        li[x] = 1
    elif cmd == "remove":
        li[x] = 0
    elif cmd == "toggle":
        li[x] = 1 - li[x]
    else:
        print(li[x])
